arches/app/models/models.py

```diff
@@ -583,7 +583,11 @@ def get_published_graph(self, language=None, raise_if_missing=False):
             )
         except PublishedGraph.DoesNotExist:
             if raise_if_missing:
-                raise
+                raise ValueError(
```

Comment:
Instead of documenting this as a minor incompatibility from 7.5 -> 7.6 (because code catching `except PublishedGraph.DoesNotExist` will no longer work), what do you think about just [enriching the exception with a note](https://docs.python.org/3.11/tutorial/errors.html#enriching-exceptions-with-notes) about the missing language?

We're doing that here: (It's a 3.11+ feature, so you'd have to add a guard for python 3.10, but we can immediately drop the guard on v8 which will be 3.11+.)
https://github.com/archesproject/arches/blob/2cf55295aff887abfb10e35c6febdac6040f2abc/arches/management/commands/load_jsonld.py#L307-L308

Reply:
Makes sense to move ahead with this given that the system check will be 8.x, but I'm still hoping we can do without changing the type of the exception.

If you don't like the add_note() idea because it won't work on all versions, you can basically fake it by recreating the exception:

```py
In [3]: from arches.app.models.models import *

In [4]: try:
   ...:     GraphModel.objects.get(pk=4)
   ...: except Exception as e:
   ...:     raise e.__class__('hello exception') from e
   ...: 
---------------------------------------------------------------------------
DoesNotExist                              Traceback (most recent call last)
Cell In[4], line 2
      1 try:
----> 2     GraphModel.objects.get(pk=4)
      3 except Exception as e:

File ~/release/lib/python3.12/site-packages/django/db/models/manager.py:87, in BaseManager._get_queryset_methods.<locals>.create_method.<locals>.manager_method(self, *args, **kwargs)
     85 @wraps(method)
     86 def manager_method(self, *args, **kwargs):
---> 87     return getattr(self.get_queryset(), name)(*args, **kwargs)

File ~/release/lib/python3.12/site-packages/django/db/models/query.py:637, in QuerySet.get(self, *args, **kwargs)
    636 if not num:
--> 637     raise self.model.DoesNotExist(
    638         "%s matching query does not exist." % self.model._meta.object_name
    639     )
    640 raise self.model.MultipleObjectsReturned(
    641     "get() returned more than one %s -- it returned %s!"
    642     % (
   (...)
    645     )
    646 )

DoesNotExist: GraphModel matching query does not exist.

The above exception was the direct cause of the following exception:

DoesNotExist                              Traceback (most recent call last)
Cell In[4], line 4
      2     GraphModel.objects.get(pk=4)
      3 except Exception as e:
----> 4     raise e.__class__('hello exception') from e

DoesNotExist: hello exception
```

Either way is fine, I just don't want to unnecessarily break code that's catching ObjectDoesNotExist.